ares/n64/pi/bus.hpp

```diff
@@ -0,0 +1,79 @@
+template<u32 Size>
+inline auto PI::busRead(u32 address) -> u64 {
+  static constexpr u64 unmapped = 0;
+
+  if (unlikely(io.ioBusy)) {
+    io.ioBusy = 0; //technically, we should wait until Queue::PI_BUS_Write
```

Comment:
Can you elaborate on why this is needed?

Reply:
As tried to describe in the comment, all reads are blocking. In case of a read while a write is ongoing, the VR4300 is stalled until the write is finished, and then the last written value is returned.

To be more accurate, we should in general emulate stalls on reads, and in a case a write is ongoing, stall the remaining amount of cycles until that write is finished.

This is a quick workaround. If you want I can call queue.remove here so that we’re more safe wrt queue overflows